src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarReader.cs

```diff
@@ -322,19 +305,15 @@ private bool TryProcessExtendedAttributesHeader(TarHeader firstHeader, bool copy
                 throw new FormatException(SR.TarTooManyGlobalExtendedAttributesEntries);
             }
 
-            // Can't have two metadata entries in a row, no matter the archive format
+            // Can't have two extended attribute metadata entries in a row
             if (secondHeader._typeFlag is TarEntryType.ExtendedAttributes)
             {
                 throw new FormatException(string.Format(SR.TarUnexpectedMetadataEntry, TarEntryType.ExtendedAttributes, TarEntryType.ExtendedAttributes));
             }
 
             Debug.Assert(firstHeader._extendedAttributes != null);
-            if (GlobalExtendedAttributes != null)
-            {
-                // First, replace some of the entry's standard attributes with the global ones
-                secondHeader.ReplaceNormalAttributesWithGlobalExtended(GlobalExtendedAttributes);
-            }
-            // Then replace all the standard attributes with the extended attributes ones,
+
+            // Replace all the standard attributes with the extended attributes ones,
             // overwriting the previous global replacements if needed
             secondHeader.ReplaceNormalAttributesWithExtended(firstHeader._extendedAttributes);
```

Comment:
Maybe this is part of the deferred work alluded to above, but if global attributes are applied in a generally forward manner why is this doing something special about if exactly the preceding entry was/had global attributes?

Reply:
Some answers for this comment:

- This method is only called when a header has been read and the entry type is `ExtendedAttributes`. It is invoked in `TryGetNextEntryHeader`.
- After analyzing the way the `pax` and `tar` unix commands work with Global Extended Attributes entries, I concluded there are two behaviors:
  - The default behavior is that a GEA will contain KVPs in its data section with any information, but it won't be propagated into the dictionaries of the subsequent entries, and they also won't affect the entries if they get extracted to disk, even if the keys have reserved names like `mode`, `devmajor`, `uid`, etc.
  - The user can also choose the option of removing the GEA entries, pasting the KVPs in the Extended Attributes dictionaries of the subsequent entries, and then writing each entry's Extended Attributes on top (because the individual ones have more priority than the ones in the GEA). But I also noticed that they do not affect extracted files either, even if they have reserved names.
- Correct, I will defer most of the GEA work to the next PR.
- This line you're talking about (339) is not going to change: Its purpose is to find if the dictionary contains keys with reserved names, in which case their values need to substitute the original values in the header, particularly because some fields have a fixed size, like `name`, `linkName`, `size`.

Reply:
I think that you're saying that the `tar` and `pax` commands both effectively just skip global extended attributes records?